### ISSUE TITLE
An option to automaterialize historical time window partitions

### ISSUE THREAD

**Issue by psarka** [dagster-io/dagster#16033]
### What's the use case?

I'm creating a daily partitioned asset with a start_date in the past that has self a partition dependency (on previous day). I would like it to auto-materialize starting from the start_date and into the future.

Currently this is explicitly not supported - documentation states "To materialize earlier partition, launch a backfill". Unfortunately this is problematic.

If I create an asset, and launch a backfill, it is possible that by the time backfill is done, there will be extra unmaterialized partitions added. And then auto-materialize scheduler will not trigger any runs.

I would like to avoid the need to "babysit" the backfills to make sure they are all fully materialized up to today. It would be nice to have an option (`AutomaterializePolicy.very_eager()`) to fill all the time window partitions for me.

### Ideas of implementation

Add another `AutomaterializePolicy` or a flag to an existing one.

### Additional information

This is a related issue: https://github.com/dagster-io/dagster/issues/11833

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@psarka is this what you're looking for? https://github.com/dagster-io/dagster/discussions/15567

**Comment by psarka** [comment]
That's exactly what I need!

Unfortunately it does not work if the asset has upstream dependencies (I have a repro here https://github.com/psarka/repro/tree/auto-time-window).

Should I file a bug for that?

Also, should the #15567 behavior be described in the docs? I could try to make a PR for that :)

**Comment by sryza** [comment]
I noticed your repro includes a self dependency. Does it also repro without a self-dependency?

> Also, should the https://github.com/dagster-io/dagster/discussions/15567 behavior be described in the docs? I could try to make a PR for that :)

That would be awesome

**Comment by paulius-hydrosat** [comment]
Without self-dependency it only materialized the last partition (despite `eager(1000)`). I made a test consisting of full matrix of options: 

dim1: [no deps, self dep, upstream dep, self dep + upstream dep]
dim2: [eager 1, eager 100]

Results are as follows:

no deps eager 1:
:heavy_check_mark: last partition

no deps eager 100:
:heavy_check_mark: all partitions in parallel

self dep eager 1:
:question: all partitions sequentially

self dep eager 100:
:heavy_check_mark: all parititions sequentially

upstream dep eager 1:
:heavy_check_mark: last partition

upstream dep eager 100:
:stop_sign: last partition

self dep upstream dep eager 1:
:question: no materializations

self dep upstream dep eager 100:
:stop_sign: no materializations

![image](https://github.com/dagster-io/dagster/assets/135983426/9b822e1c-0e86-4e77-8253-ab7bb1de1748)

**Comment by paulius-hydrosat** [comment]
(Also updated the repro repository)
